docs/advanced/experimental/Kubernetes.md

```diff
@@ -0,0 +1,132 @@
+# Cookbook on `Kubernetes` Deployment
+
+Deploy your `Flow` on `Kubernetes`.
+## Requirements
+- please setup a `Kubernetes` cluster and pull the credentials. e.g.:
+  [minikube](https://minikube.sigs.k8s.io/docs/start/) (testing),
+  [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine),
+  [Amazon EKS](https://aws.amazon.com/eks),
+  [Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service),
+  [Digital Ocean](https://www.digitalocean.com/products/kubernetes/)
+  
+## Deploy your `Flow`
+To deploy a `Flow` on `Kubernetes`, you have to set the `infrastructure='K8S'` when creating the `Flow`.
+Call `flow.start()` to trigger the deployment. Make sure all `Executors` are containerized.
+
+## Example
+In this example, we assume that, there is a postgres database deployed on `Kubernetes` already. 
+You can also modify the database configuration to use a postgres database outside the `Kubernetes` cluster.
+Deploy the index `Flow` on `Kubernetes`:
+```python
+from jina import Flow
+
+flow = Flow(
+    name='index-flow', port_expose=8080, infrastructure='K8S', protocol='http'
+).add(
+    # name of the service and deployment in Kubernetes
+    name='test_searcher',
+    # executor has to be containerized
+    uses='jinahub+docker://FaissPostgresSearcher',
+    # database configuration
+    uses_with={
+        'hostname': f'postgres.postgres.svc.cluster.local',
+        'username': 'postgresadmin',
+        'database': 'postgresdb',
+        'table': 'test_searcher',
+    }
+)
+flow.start()
```

Comment:
can this flow be used in the context manager style? 

```python
with flow:
   ...
```
As `start()` is not introduced at all in Flow docs, and where is K8s deconstructor called?

Reply:
I think there is no destructor yet as far as I am aware. The namespace has to be destroyed manually atm.

Reply:
Here is a ticket to support that feature:
https://github.com/jina-ai/jina/issues/3334